src/deepsparse/v2/text_generation/pipeline.py

```diff
@@ -34,10 +34,63 @@
     ProcessOutputs,
     TokenGeneratorOperator,
 )
+from deepsparse.v2.transformers.pipeline import TransformersPipeline
 from deepsparse.v2.utils import PipelineState
 
 
-class TextGenerationPipeline(Pipeline):
+class TextGenerationPipelineNoCache(TransformersPipeline):
+    def __init__(
+        self,
+        model_path: str,
+        sequence_length: int = 1024,
+        engine_kwargs: Optional[Dict] = None,
+        onnx_model_name: Optional[str] = None,
+        **kwargs,
+    ):
+
+        self.model_path = self.setup_onnx_file_path(
```

Comment:
Line 51 to 56 should be ideally executed by the parent of `TextGenerationPipelineNoCache` and `TextGenerationPipeline` - the `TransformersPipeline`.

However, we encounter a bit of a pickle, since we need to pass down the "inheritance chain" ops, router and schedulers, so we cannot call `__init__` before we have those variables in place. This is not an issue for `TextGenerationPipelineNoCache`, but is an issue for `TextGenerationPipeline`. The latter needs to do a lot of class-specific work between calling `self.setup_onnx_file_path` and calling `super().__init__`. Let's think about whether we can improve the implementation.